### PR TITLE
Add test for `LabelEncoder` with `add_noise=True` for 'category' pandas dtype

### LINKED ISSUE #617: LabelEncoder() with add_noise = True does not Support 'category' pandas dtype
### Problem Description
Currently, the `LabelEncoder(add_noise = True)` does not support 'category' pandas dtype. For instance, if I run the following code:
```
from rdt.transformers import LabelEncoder
data_test = pd.DataFrame({'A': ['a', 'b', 'a', 'a', 'c']})
data_test = data_test.astype('category')
transformer = HyperTransformer()
transformer.set_config({
  'sdtypes': {
    'A': "categorical"
  },
  'transformers': {
    'A': LabelEncoder(add_noise=True),
  } 
})
transformer.fit(data_test)
transformer.transform(data_test)
```
I get the error: 
`TypeError: unsupported operand type(s) for +: 'Categorical' and 'int'`

This comes from these lines in `rdt/transformers/categorical.py`:
```
if self.add_noise:
--> 527             mapped = np.random.uniform(mapped, mapped + 1)

```
### Expected behavior
Supporting 'category' pandas dtype.

### Additional context

However the code above works with `FrequencyEncoder(add_noise=True/False)` and `LabelEncoder(add_noise=False)`

### PR DESCRIPTION
**Pull request by fealho**
CU-860q1dv6r, Resolve #617.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-860q1dv6r RDT - LabelEncoder() with add_noise = True does not Support 'category' pandas dtype #617](https://app.clickup.com/t/860q1dv6r)

**Review by amontanez24 — COMMENTED** [review]
Just to sanity check can we reverse transform in the test as well?

File: tests/integration/transformers/test_categorical.py [hunk 1]
```diff
@@ -665,6 +666,35 @@ def test_label_encoder_order_by_alphabetical():
     pd.testing.assert_frame_equal(reverse, data)
 
 
+def test_label_encoder_add_noise():
+    """Test the LabelEncoder with ``add_noise``."""
+    # Setup
+    data_test = pd.DataFrame({'A': ['a', 'b', 'a', 'a', 'c']})
+    data_test = data_test.astype('category')
```
Comment (gsheni) [inline tests/integration/transformers/test_categorical.py:673]:
```suggestion
    data_test = pd.DataFrame({'A': pd.Series(['a', 'b', 'a', 'a', 'c'], dtype='category' )})
```

**Review by gsheni — CHANGES_REQUESTED** [review]

File: tests/integration/transformers/test_categorical.py (hunk 1 above)
Comment (gsheni) [reply tests/integration/transformers/test_categorical.py:673]: Also, lint is failing

**Review by amontanez24 — APPROVED** [review]

**Review by gsheni — APPROVED** [review]

**Pull request merged**